File: OLSource1/v1/VS_officedev/codesnippet/CSharp/walkthrough--binding-to-data-from-a-service-in-a-vsto-add-in-project_4.cs
        void richTextContentControl_Entering(object sender, ContentControlEnteringEventArgs e)
        {
            document[0] = new ContentService.requestedDocument();
            document[0].type = ContentService.documentTypes.primary;
            document[0].selector = "Mtps.Xhtml";

            request.contentIdentifier = "ed59e522-dd6e-4c82-8d49-f5dbcfcc950d";
            request.locale = "en-us";
            request.version = "VS.90";
            request.requestedDocuments = document;

            response = proxy.GetContent(appId, request);
            primaryDocumentsBindingSource.DataSource =
                response.primaryDocuments[0].Any.InnerText;
            richTextContentControl.DataBindings.Add("Text",
                primaryDocumentsBindingSource.DataSource, "", true,
                System.Windows.Forms.DataSourceUpdateMode.OnValidation);
        }

File: OLSource1/v1/VS_officedev/codesnippet/VisualBasic/walkthrough--binding-to-data-from-a-service-in-a-vsto-add-in-project_4.vb
    Private Sub richTextContentControl_Entering _
        (ByVal sender As Object, ByVal e As ContentControlEnteringEventArgs) _
        Handles richTextContentControl.Entering

        document(0) = New ContentService.requestedDocument()
        With document(0)
            .type = ContentService.documentTypes.primary
            .selector = "Mtps.Xhtml"
        End With

        With request
            .contentIdentifier = "ed59e522-dd6e-4c82-8d49-f5dbcfcc950d"
            .locale = "en-us"
            .version = "VS.90"
            .requestedDocuments = document
        End With

        response = proxy.GetContent(appId, request)

        primaryDocumentsBindingSource.DataSource = _
            response.primaryDocuments(0).Any.InnerText
        richTextContentControl.DataBindings.Add("Text", _
            primaryDocumentsBindingSource.DataSource, "", True, _
            System.Windows.Forms.DataSourceUpdateMode.OnValidation)
    End Sub